src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -342,6 +349,15 @@ private String buildOperationName(Service service, ExecutionContext executionCon
     private NadelContext getNadelContext(ExecutionContext executionContext) {
         return executionContext.getContext();
     }
+
+    private boolean skipTransformationProcessing(NadelContext nadelContext, QueryTransformationResult transformedQuery) {
+        TransformationState transformations = transformedQuery.getTransformations();
+        return nadelContext.getNadelExecutionHints().isOptimizeOnNoTransformations() &&
+                transformations.getFieldIdToTransformation().size() == 0 &&
+                transformations.getTypeRenameMappings().size() == 0 &&
+                !transformedQuery.getRemovedFieldMap().hasRemovedFields() &&
+                transformations.getHintTypenames().size() == 0;
+    }
```

Comment:
I am happy to merge this but for future reference, this is an lot to unpack as a reader.

I personally like a big comment that explains the reasons for this.  Not explains the code line by line but explains the  theory behind the code here.  

Anyway carry on